CHANGELOG.md

```diff
@@ -287,6 +287,14 @@ The currently supported dialects are `mysql-dialect`, `postgresql-dialect`, `hsq
 - [Compiler] Emit annotations on properties instead of types (#2798 by [Alexander Perfilyev][aperfilyev])
 - [Compiler] Sort arguments before passing to a Query subtype (#2379 by [Alexander Perfilyev][aperfilyev])
 
+## [1.5.5] - 2023-01-20
+
+This is a compatibility release for Kotlin 1.8 and IntelliJ 2021+, supporting JDK 17.
```

Comment:
What you have here is probably all that is needed to note in the changelog, but for posterity I feel like I should jot this more detailed explanation down somewhere public:

1.5.4 also works with JDK 17 so long as you are using the gradle daemon. Gradle [adds `--add-opens` flags](https://github.com/gradle/gradle/blob/v7.5.0/subprojects/base-services/src/main/java/org/gradle/internal/jvm/JpmsConfiguration.java#L35) to the JVM when forking daemon processes that will suppress the illegal reflection that happens in the IntelliJ library. Our builds were disabling the gradle daemon, which is why they went 💥 

Sometimes Gradle still starts a daemon process when `org.gradle.daemon=false`. This happens when the JVM settings passed to `org.gradle.jvmargs` conflict with the settings that the client process was started with. For example, if `org.gradle.jvmargs` specifies `-Xmx30g`, this is much more heap space than the client process allocates by default, so a single-use daemon is spawned instead, and shutdown at the end of the build. You will see this in your logs when that happens: 

```
To honour the JVM settings for this build a single-use Daemon process will be forked
```

If you instead pass `-Xmx30g` to `JAVA_OPTS` when invoking the Gradle wrapper, the client process now has enough heap to execute the build in-process without forking a new JVM. In this case, the default gradle daemon add-opens flags are not applied to the client process JVM, which is what caused our builds to fail.

Gradle has been [removing implicit add-opens flags](https://docs.gradle.org/current/userguide/upgrading_version_7.html#remove_implicit_add_opens_for_gradle_workers) from the daemon workers in 8.x, but at the time of writing they still have opens for `java.base/java.util`. Gradle also intends on [getting rid of the org.gradle.daemon=false option](https://github.com/gradle/gradle/issues/17229#issuecomment-863232813), and will always spawn a daemon process to execute builds.

Older versions using the same IntelliJ library as 1.5.4 may also work with JDK 17 but I did not bother to test them.

Reply:
Thanks for the nice details, but I think this is too detailed for the changelog.